src/index.js

```diff
@@ -9,27 +9,39 @@ const { Template } = require("./messages/template");
 const Text = require("./messages/text");
 
 const api = require("./fetch");
+const { post, get } = require("./requests");
+
+const EventEmitter = require("node:events")
+const { createHmac } = require('node:crypto');
 
 /**
  * The main API object
  *
  * @property {String} token The API token
+ * @property {String} appSecret The app secret
+ * @property {String} webhookVerifyToken The webhook verify token
  * @property {String} v The API version to use
  * @property {Boolean} parsed If truthy, API operations will return the fetch promise instead. Intended for low level debugging.
  */
-class WhatsAppAPI {
+class WhatsAppAPI extends EventEmitter {
     /**
      * Initiate the Whatsapp API app
      *
      * @param {String} token The API token, given at setup. It can be either a temporal token or a permanent one.
+     * @param {String} appSecret The app secret, given at setup.
+     * @param {String} webhookVerifyToken The webhook verify token, given at setup.
      * @param {String} v The version of the API, defaults to v14.0
      * @param {Boolean} parsed Whether to return a pre-processed response from the API or the raw fetch response. Intended for low level debugging.
      * @throws {Error} If token is not specified
      */
-    constructor(token, v = "v15.0", parsed = true) {
+    constructor(token, appSecret = "", webhookVerifyToken = "", v = "v15.0", parsed = true) {
```

Comment:
Rather than defaulting to empty strings, I think they should be set as mandatory.

It's either that or changing the 404-411 lines to be optional, which I actually prefer since the check is optional in the API.

Reply:
I wouldn't break the call `new WhatsAppAPI(token)` just to show you the new functionality. I think you should decide which syntax will be.

Regarding 404-411 it's a potential vulnerability not to use validation. I would protect users as default.

P.S. In common, I prefer named arguments for functions using objects, rather than positional ones.

Reply:
Yeah, I had been thinking about it the last two days, specially for the OnStatus and OnMessage. Now that the class has more arguments, I think it's time to change it.

I will do it in a PR later.